### ISSUE TITLE
[BUG] Unexpected 500 Error Response When Using DEV Client CLI with LTS Docker Compose

### ISSUE THREAD

**Issue by astrokov7** [hyperledger/iroha#3570]
### OS and Environment

MacOS, Docker Hub

### Branch

Last LTS

### Minimum working example / Steps to reproduce

1. Set up and start Iroha 2 using the LTS version of Docker Compose.
2. Manage Iroha using the Dev Client CLI.
3. ```./iroha_client_cli domain list all```
4. Observe the error response.

### Actual result

```
Error: 
   0: Failed to get all domains
   1: Other error
   2: Unexpected query response; status: 500 Internal Server Error; response body: Unhandled rejection: WarpQueryError(Decode(ParityScale))
```

### Expected result

A 400-series error indicating the client version is deprecated should be returned when using an outdated client with a newer server version.

### Who can help to reproduce?

@astrokov7

### Notes

Implement error handling that checks the version compatibility between the client and server, returning an appropriate error message when a deprecated client is used. This can guide users to update their client version for compatibility with the server.

**Comment by Erigara** [comment]
Already implemented on `iroha2-dev` branch, opened #3602 to improve error messages.

**Comment by timofeevmd** [comment]
done
